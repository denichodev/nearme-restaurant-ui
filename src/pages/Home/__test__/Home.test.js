import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { Home } from '../index';

describe('<Home />', () => {
  it('renders correctly', () => {
    const props = {
      favorite: {
        loading: false,
        data: [
          {
            additionalInfo: [],
            address:
              'Gandaria City, Lantai Upper Ground, Jl. Sultan Iskandar Muda, Gandaria, Jakarta',
            cover: 'http://localhost:8080/assets/top-momoiro.jpg',
            cuisine: 'Asian',
            id: 1,
            location: 'Kota Kasablanka, Tebet, Jakarta',
            name: 'Momoiro',
            openingHours: '10.00 - 22.00',
            phoneNumber: '08980780780',
            rating: 4.2,
            slides: [
              'http://localhost:8080/assets/momoiro-1.jpg',
              'http://localhost:8080/assets/momoiro-2.jpg',
              'http://localhost:8080/assets/momoiro-3.jpg',
            ],
            slug: 'momoiro-kota-kasablanka',
          },
        ],
      },
      getFavoriteFood: jest.fn(),
    };
    const wrapper = shallow(<Home {...props} />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
