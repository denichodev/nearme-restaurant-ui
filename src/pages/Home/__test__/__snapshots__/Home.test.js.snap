// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<div
  className="hero"
>
  <Grid
    className=""
    fluid={false}
  >
    <Hero
      data={
        Array [
          Object {
            "additionalInfo": Array [],
            "address": "Gandaria City, Lantai Upper Ground, Jl. Sultan Iskandar Muda, Gandaria, Jakarta",
            "cover": "http://localhost:8080/assets/top-momoiro.jpg",
            "cuisine": "Asian",
            "id": 1,
            "location": "Kota Kasablanka, Tebet, Jakarta",
            "name": "Momoiro",
            "openingHours": "10.00 - 22.00",
            "phoneNumber": "08980780780",
            "rating": 4.2,
            "slides": Array [
              "http://localhost:8080/assets/momoiro-1.jpg",
              "http://localhost:8080/assets/momoiro-2.jpg",
              "http://localhost:8080/assets/momoiro-3.jpg",
            ],
            "slug": "momoiro-kota-kasablanka",
          },
        ]
      }
      loading={false}
    />
  </Grid>
</div>
`;
